File: ando_toolkit/src/main/java/com/ando/toolkit/NumberUtils.java
package com.ando.toolkit;

import java.math.BigDecimal;

/**
 * Title:NumberUtils
 * <p>
 * Description:数据工具类
 * </p>
 *
 * @author javakam
 * Date 2018/8/17 14:40
 */
public class NumberUtils {

    private static float setScale(float value, int scale) {
        BigDecimal decimal = new BigDecimal(value);
        return decimal.setScale(scale, BigDecimal.ROUND_HALF_UP).floatValue();
    }

    private static double setScale(double value, int scale) {
        BigDecimal decimal = new BigDecimal(value);
        return decimal.setScale(scale, BigDecimal.ROUND_HALF_UP).doubleValue();
    }

    public static int parseInteger(String value) {
        int result = 0;
        if (value != null) {
            try {
                return Integer.parseInt(value);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return result;
    }

    public static int parseInteger(Number value) {
        int result = 0;
        if (value != null) {
            return value.intValue();
        }
        return result;
    }

    public static long parseLong(String value) {
        long result = 0;
        if (value != null) {
            try {
                return Long.parseLong(value);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return result;
    }

    public static long parseLong(Number value) {
        long result = 0;
        if (value != null) {
            return value.longValue();
        }
        return result;
    }

    public static float parseFloat(String value, int scale) {
        float result = 0.00f;
        if (value != null) {
            try {
                result = Float.parseFloat(value);
                return setScale(result, scale);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return result;
    }

    public static float parseFloat(Number value, int scale) {
        float result = 0.00f;
        if (value != null) {
            return setScale(value.floatValue(), scale);
        }
        return result;
    }

    public static double parseDouble(String value, int scale) {
        double result = 0.00f;
        if (value != null) {
            try {
                result = Double.parseDouble(value);
                return setScale(result, scale);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return result;
    }

    public static double parseDouble(Number value, int scale) {
        double result = 0.00f;
        if (value != null) {
            return setScale(value.doubleValue(), scale);
        }
        return result;
    }


}

File: ando_toolkit/src/main/java/com/ando/toolkit/NumberUtils.kt
package com.ando.toolkit

import java.math.BigDecimal

/**
 * Title:NumberUtils
 *
 *
 * Description:数据工具类
 *
 *
 * @author javakam
 * Date 2018/8/17 14:40
 */
object NumberUtils {
    private fun setScale(value: Float, scale: Int): Float {
        val decimal: BigDecimal = BigDecimal(value)
        return decimal.setScale(scale, BigDecimal.ROUND_HALF_UP).toFloat()
    }

    private fun setScale(value: Double, scale: Int): Double {
        val decimal = BigDecimal(value)
        return decimal.setScale(scale, BigDecimal.ROUND_HALF_UP).toDouble()
    }

    fun parseInteger(value: String?): Int {
        val result = 0
        if (value != null) {
            try {
                return value.toInt()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        return result
    }

    fun parseInteger(value: Number?): Int {
        val result = 0
        return value?.toInt() ?: result
    }

    fun parseLong(value: String?): Long {
        val result: Long = 0
        if (value != null) {
            try {
                return value.toLong()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        return result
    }

    fun parseLong(value: Number?): Long {
        val result: Long = 0
        return value?.toLong() ?: result
    }

    fun parseFloat(value: String?, scale: Int): Float {
        var result = 0.00f
        if (value != null) {
            try {
                result = value.toFloat()
                return setScale(result, scale)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        return result
    }

    fun parseFloat(value: Number?, scale: Int): Float {
        val result = 0.00f
        return if (value != null) {
            setScale(value.toFloat(), scale)
        } else result
    }

    fun parseDouble(value: String?, scale: Int): Double {
        var result = 0.00
        if (value != null) {
            try {
                result = value.toDouble()
                return setScale(result, scale)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        return result
    }

    fun parseDouble(value: Number?, scale: Int): Double {
        val result = 0.00
        return if (value != null) {
            setScale(value.toDouble(), scale)
        } else result
    }
}